Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1683.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System;
+
+using Xamarin.Forms.CustomAttributes;
+using Xamarin.Forms.Internals;
+using System.Collections.Generic; 
+
+#if UITEST
+using Xamarin.UITest;
+using NUnit.Framework;
+using System.Diagnostics;
+#endif
+
+namespace Xamarin.Forms.Controls.Issues
+{
+	[Preserve (AllMembers = true)]
+	[Issue (IssueTracker.Github, 1683, "Auto Capitalization Implementation")]
+	public class Issue1683 : ContentPage
```

Comment:
Can we maybe add a UITest to this that just rotates through all the options and captures screenshots? I'm not sure if it can actually check for the capitalization changes, but I'd like to at least have it run on devices to catch any potential crashes.

Reply:
Also, maybe add to this test a custom renderer that just sets the capitalization. Does the custom renderer still work if we haven't set the `KeyboardFlags`?

Reply:
I started down the path of the UITests but then realized the only way to test these would be to type onto the keyboard which AFAICT you can't really do. But your second point is a good one so I'll just add the UITests I wrote back in to just cycle through the fields and make sure it doesn't crash.

I'll added a custom renderer version of these as well and test through them